ci/travis/ci.sh

```diff
@@ -38,21 +38,142 @@ should_run_job() {
   return "${skip}"
 }
 
+# Idempotent environment loading
 reload_env() {
   # TODO: We should really just use a new login shell instead of doing this manually.
   # Otherwise we might source a script that isn't idempotent (e.g. one that blindly prepends to PATH).
-  { local set_x="${-//[^x]/}"; } 2> /dev/null  # save set -x to suppress noise
-  set +x
-  local to_add="$HOME/miniconda/bin" old_path=":${PATH}:"
-  if [ "${old_path}" = "${old_path##*:${to_add}:*}" ]; then
-    PATH="${to_add}:$PATH"
+
+  if [ -n "${GITHUB_PULL_REQUEST-}" ]; then
+    case "${GITHUB_PULL_REQUEST}" in
+      [1-9]*) TRAVIS_PULL_REQUEST="${GITHUB_PULL_REQUEST}";;
+      *) TRAVIS_PULL_REQUEST=false;;
+    esac
+    export TRAVIS_PULL_REQUEST
+  fi
+
+  export PYTHON3_BIN_PATH=python
+  export GOROOT="${HOME}/go" GOPATH="${HOME}/go_dir"
+  if [ "${OSTYPE}" = msys ]; then
+    export USE_CLANG_CL=1
+  fi
+
+  # NOTE: Modifying PATH invalidates Bazel's cache! Do not add to PATH unnecessarily.
+  PATH="${HOME}/miniconda/bin":"${PATH}"
+  if [ "${OSTYPE}" = msys ]; then
+    PATH="${HOME}/miniconda/bin/Scripts":"${PATH}"
   fi
-  test -z "${set_x}" || set -x  # restore set -x
+
+  # Deduplicate PATH
```

Comment:
This should not be needed...

Reply:
What are you basing this on?

`reload_env` can (and I believe is) called multiple times in a single shell. If you change `PATH` *n* times you're going to build with Bazel *n* times. I don't think that's a good idea.